client/lib/plans/index.js

```diff
@@ -67,7 +67,33 @@ export function addCurrentPlanToCartAndRedirect( sitePlans, selectedSite ) {
 }
 
 export function getCurrentPlan( plans ) {
-	return find( plans, { currentPlan: true } );
+	const currentPlan = find( plans, { currentPlan: true } );
+
+	if ( currentPlan ) {
+		return currentPlan;
+	}
+
+	// the site doesn't have defined a current plan.
+	// Let's dive in its `plan` property
+	const site = sitesList.getSelectedSite();
+
+	if ( ! site ) {
+		return null;
+	}
+
+	const { plan } = site;
+
+	if ( ! plan ) {
+		return null;
+	}
+
+	// let's check if the current plan is the old `Enterprise` plan.
+	// if it is then let's assing `business` plan.
+	if ( plan.product_id === 1007 ) {
```

Comment:
This seems fragile to be hard-coding this number here. Would it not be better to add this into the `isEnterprise()` function in [lib/products-values/index.jsx](https://github.com/Automattic/wp-calypso/blob/master/client/lib/products-values/index.js#L95) or wherever that function gets its information? A named-constant maybe…